backend/src/route.rs

```diff
@@ -233,12 +233,24 @@ mod tests {
         //               |
         //              i0
         // ```
-        let map = load_osm_xml("simple_four_way_intersection");
+        let mut map = load_osm_xml("simple_four_way_intersection");
         let Route { steps } = map.router_before.route_from_roads(r(3), r(2)).unwrap();
         assert_eq!(
             steps,
             vec![(r(3), Direction::Backwards), (r(2), Direction::Backwards)]
         );
+
+        // Filter r2
+        map.add_modal_filter(
```

Comment:
Thank you for adding a test! I think routing could be a bit of a whack-a-mole problem, so it's good to make sure we're not regressing.

Reply:
Easy to add tests once there's at least one :)  Let the virtuous cycle continue!